src/ess/reduce/nexus/workflow.py

```diff
@@ -389,6 +389,7 @@ def assemble_detector_data(
     grouped = nexus.group_event_data(
         event_data=event_data, detector_number=detector.coords['detector_number']
     )
+    grouped.coords.pop('event_id')
```

Comment:
I think this should be done in `group_event_data` because that function adds `event_id` as an auxiliary coord. So it should never escape that function.

Reply:
I was thinking the opposite. If it is removed, how to tell what `event_id` the events in each bin are associated with? Removing the coord makes the return value of `group_event_data` useless unless the caller is knowledgeable enough to be aware that they need to re-add the `detector_number` coord.

Reply:
I think we should rename `event_id` to `detector_number` before returning from `group_event_data`.